src/shared/src/AuditLogDynamoGateway/CalculateMessageStatusUseCase.ts

```diff
@@ -0,0 +1,112 @@
+import type { AuditLogEvent } from "shared-types"
+import { AuditLogStatus, EventType } from "shared-types"
+
+export default class CalculateMessageStatusUseCase {
+  private readonly events: AuditLogEvent[] = []
+
+  constructor(...allEvents: (AuditLogEvent[] | AuditLogEvent)[]) {
+    this.events = allEvents
+      .flatMap((events: AuditLogEvent[] | AuditLogEvent) => (Array.isArray(events) ? events : [events]))
+      .sort((eventA, eventB) => (eventA.timestamp > eventB.timestamp ? 1 : -1))
+  }
+
+  call(): string {
+    let status = AuditLogStatus.processing
```

Comment:
Nit: Instead of declaring `status` here, couldn't you just return each `AuditLogStatus` within the if statements?